CMakeLists.txt

```diff
@@ -203,6 +203,34 @@ if(ENABLE_LIBPNG_SUPPORT)
     list(APPEND LIBRARY_SOURCES Auxiliary/DirectXTexPNG.cpp)
 endif()
 
+macro(call_mingw_fxc work_dir input_filename output_filename entry_point var_name semantic _targetname)
```

Comment:
Why not put these macros into your .cmake?

Reply:
Because we have to track what files are built so we can setup the appropriate dependencies.   The macros just save a little typing.